plugins/test/framework.h

```diff
@@ -28,10 +29,51 @@
 
 namespace service_extensions_samples {
 
+// Parameters to customize Context behaviors.
+struct ContextOptions {
+  // Wasm logging output file.
+  std::ofstream log_file;
+  // Static time returned to wasm.
+  absl::Time clock_time = absl::UnixEpoch();
+};
+
+// Buffer class to handle copying to and from an individual BODY chunk.
+class Buffer : public proxy_wasm::BufferBase {
+ public:
+  Buffer() = default;
+
+  // proxy_wasm::BufferInterface
+  size_t size() const override;
+  proxy_wasm::WasmResult copyTo(proxy_wasm::WasmBase* wasm, size_t start,
+                                size_t length, uint64_t ptr_ptr,
+                                uint64_t size_ptr) const override;
+  proxy_wasm::WasmResult copyFrom(size_t start, size_t length,
+                                  std::string_view data) override;
+
+  // proxy_wasm::BufferBase
+  void clear() override {
+    proxy_wasm::BufferBase::clear();
+    owned_string_buffer_ = "";
+  }
+  Buffer* set(std::string_view data) {
+    proxy_wasm::BufferBase::set(data);
+    return this;
+  }
```

Comment:
I don't see the need to override the implementations of these methods that BufferBase provides.

The redefinition of set() also opens the door to possible problems: since BufferBase does not declare set() as virtual (which it probably should), the declaration of set() here doesn't get the dynamic dispatch that you would normally expect with a virtual method. IOW, if you invoke buffer->set(...) where buffer is a Buffer*, Buffer::set() will be called, but if buffer is a BufferBase*, BufferBase::set() will be called.